internal/cmd/buildtool/generic.go

```diff
@@ -55,3 +64,20 @@ func genericBuildPackage(deps buildtoolmodel.Dependencies, product *product) {
 
 	runtimex.Try0(shellx.RunEx(defaultShellxConfig(), argv, &shellx.Envp{}))
 }
+
+// genericBuildLibrary is the generic function for building a library.
+func genericBuildLibrary(deps buildtoolmodel.Dependencies, product *product) {
```

Comment:
can you write tests for this functionality like we have tests for other generic builds?